### PR TITLE
fix(cdc): fix drop source for cdc source job

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
- record dependence in `table.dependent_relations` for table-on-source
- drop internal tables correctly

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/manager/catalog/mod.rs [hunk 1]
```diff
@@ -1274,9 +1274,13 @@ impl CatalogManager {
                         continue;
                     }
 
-                    // add cdc source id
+                    // cdc source streaming job
```
Comment (chenzl25) [inline src/meta/src/manager/catalog/mod.rs:1277]: Any reason why we need an additional `all_cdc_source_ids ` instead of `all_source_ids`?

**Review by chenzl25 — COMMENTED** [review]
Any reason why we need an additional `all_cdc_source_ids ` instead of `all_source_ids`?

File: src/meta/src/manager/catalog/mod.rs (hunk 1 above)
Comment (StrikeW) [reply src/meta/src/manager/catalog/mod.rs:1277]: `all_cdc_source_ids` is for recording stream job id for the cdc source job, then caller will call `drop_streaming_jobs` to drop actors for cdc source job.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**